functional/pipeline.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, sequence, transform=None):
             self._base_sequence = sequence._unwrap_sequence()
             self._lineage = Lineage(prior_lineage=sequence._lineage)
         elif isinstance(sequence, list) or isinstance(sequence, tuple) or is_iterable(sequence):
+            print(type(sequence))
```

Comment:
Oh, my bad, somehow I completely missed that when checking the diff. Will correct in ~10 hours.